wrappers/services_test.go

```diff
@@ -3180,6 +3221,60 @@ apps:
 	})
 }
 
+func (s *servicesTestSuite) TestRestartInDifferentStates(c *C) {
+	const manyServicesYaml = `name: test-snap
+version: 1.0
+apps:
+  svc1:
+    command: bin/foo
+    daemon: simple
+  svc2:
+    command: bin/foo
+    daemon: simple
+  svc3:
+    command: bin/foo
+    daemon: simple
+  svc4:
+    command: bin/foo
+    daemon: simple
+`
+	srvFile1 := "snap.test-snap.svc1.service"
+	srvFile2 := "snap.test-snap.svc2.service"
+	srvFile3 := "snap.test-snap.svc3.service"
+	srvFile4 := "snap.test-snap.svc4.service"
+
+	info := snaptest.MockSnap(c, manyServicesYaml, &snap.SideInfo{Revision: snap.R(1)})
+
+	r := systemd.MockSystemctl(func(cmd ...string) ([]byte, error) {
+		states := map[string]serviceState{
+			srvFile1: serviceState{"active", "enabled"},
+			srvFile2: serviceState{"inactive", "enabled"},
+			srvFile3: serviceState{"active", "disabled"},
+			srvFile4: serviceState{"inactive", "disabled"},
+		}
+		if out := mockSystemCtlShow(cmd, states); out != nil {
+			return out, nil
+		}
+		s.sysdLog = append(s.sysdLog, cmd)
+		return []byte("ActiveState=inactive\n"), nil
+	})
+	defer r()
+
+	err := wrappers.AddSnapServices(info, nil, progress.Null)
+	c.Assert(err, IsNil)
+
+	s.sysdLog = nil
+	c.Assert(wrappers.RestartServices(info.Services(), nil, progress.Null, s.perfTimings), IsNil)
+	c.Check(s.sysdLog, DeepEquals, [][]string{
+		{"stop", srvFile1},
+		{"show", "--property=ActiveState", srvFile1},
```

Comment:
I'd expect Id,ActiveState,UnitFileState,Type to also be queried because afaict this is how Status() works (and also mockSystemCtlShow expects that above)?

Reply:
Yes, but if you see how I'm setting up the mock, I'm not logging that call.

Reply:
Ah, sorry, missed that, and "show", "--property=ActiveState" comes from stop itself, that's what confused me.